src/applications/ueye_installation_check.cpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+* SOFTWARE LICENSE AGREEMENT (BSD LICENSE):
+*
+* Copyright (c) 2016, Kei Okada, Daniel Stonier
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions
+* are met:
+*
+*  * Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+*  * Redistributions in binary form must reproduce the above
+*    copyright notice, this list of conditions and the following
+*    disclaimer in the documentation and/or other materials provided
+*    with the distribution.
+*  * Neither the name of the School of Computer Science, McGill University,
+*    nor the names of its contributors may be used to endorse or promote
+*    products derived from this software without specific prior written
+*    permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <dlfcn.h>
+#include <iostream>
+#include <sstream>
+
+#include <tclap/CmdLine.h>
+#include <tclap/SwitchArg.h>
+
+#include <algorithm>
+#include <set>
+
+int main(int argc, char** argv) {
+
+    try {
```

Comment:
Is TCLAP really doing much for us here in checking the installation?

Reply:
I'm a big fan of self-documenting programs. Even if it's just the ability to run `-h` on them.

Admittedly, I was planning on building these out more substantially (at which point it would be obtuse to not use a command line option-like library), but ended up not doing so.

I can also see the benefits of dropping dependencies too. TCLAP is pretty light though (header only), and I think the only time I'd care is if I'd be compiling this code for an esoteric embedded platform. It's already a fully fledged ROS package though, with all the dependencies that brings in, so not likely.